### ISSUE TITLE
Tag Parser incorrectly identifies the type of multiple pointers declared in one line

### ISSUE THREAD

**Issue by tylertian123** [microsoft/vscode-cpptools#2825]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 Home 10.0.17134 Build 17134
- VS Code Version: 1.29
- C/C++ Extension Version: 0.20.1
- Other extensions you installed (and if the issue persists after disabling them): Code Spell Checker, Debugger for Java, Gradle Language Support, Java Extension Pack, Java Test Runner, Language Support for Java by Red Hat, Markdown Preview GitHub Styling, Maven for Java, Paste and Indent, Python.

After `#include`ing a file that could not be found, which puts the Tag Parser in charge of providing IntelliSense features, the types of member variables `b` and `c` in the below code (which pop up when the mouse is hovered over them) are incorrectly identified.

```cpp
#include "nonexsistent.h"

class A {
};
class B {
    A *a, *b, c;
}
```

**To Reproduce**

1. Create a new file and save as C++
1. Enter the code mentioned before
1. Hover over the names of member variables `B::b` and `B::c`
1. Error: `b` is identified as a pointer-to-pointer and `c` is identified as a pointer
**Expected behavior**

The correct type of `B::b` should be `A*` and not `A**`, and the correct type of `B::c` should be `A` and not `A*`.

**Screenshots**

![image](https://user-images.githubusercontent.com/32781310/48516196-bfdff580-e830-11e8-8177-891e8b026ce7.png)
![image](https://user-images.githubusercontent.com/32781310/48516174-af2f7f80-e830-11e8-839a-9e7df1d388dc.png)

**Comment by sean-mcmanus** [comment]
Yeah, our parser is storing the incorrect types in the database. Visual Studio has the same underlying bug, but they don't expose a "Tag Parser" mode for hover.

**Comment by sean-mcmanus** [comment]
I filed a Visual Studio bug at https://developercommunity.visualstudio.com/content/problem/383095/c-class-views-shows-the-incorrect-types-for-a-clas.html . The more people that upvote that the more likely it is to get fixed. Right now, it's probably a low priority issue for them.

**Comment by Colengms** [comment]
This appears to have been addressed some time ago, and no longer repros in 1.6.0.  Closing as fixed.
